pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -404,8 +408,14 @@ func CallExtenders(extenders []framework.Extender, pod *v1.Pod, nodeLister frame
 
 	var newCandidates []Candidate
 	for nodeName := range victimsMap {
+		// check if victims.Pods is empty
+		victims := victimsMap[nodeName]
+		if len(victims.Pods) == 0 {
```

Comment:
I'd prefer to run the check around L391: if an extender returns illegal result and `extender.IsIgnorable() == false`, return it as an error.

Reply:
> I'd prefer to run the check around L391: if an extender returns illegal result and `extender.IsIgnorable() == false`, return it as an error.

Check the validness of `nodeNameToVictims `returned by `PrecessPreemption`  before L399?
```
for nodeName := range nodeNameToVictims {
	// check if victims.Pods is empty
 		victims := victimsMap[nodeName]
 		if len(victims.Pods) == 0 {
 		    delete(victims, nodeName)
 		 }
}  
```

Reply:
before L399, like this:

```go
		for nodeName, victims := range nodeNameToVictims {
			if victims != nil && len(victims.Pods) != 0 {
				continue
			}
			if extender.IsIgnorable() {
				delete(nodeNameToVictims, nodeName)
				// klog....
				continue
			} else {
				return nil, framework.AsStatus(/* build an internal err */)
			}
		}
```